src/store/model/helpers/nvidia.ts

```diff
@@ -7,10 +7,17 @@ import open from 'open';
 import {timestampUrlParameter} from '../../timestamp-url-parameter';
 
 function getRegionInfo(): NvidiaRegionInfo {
-	const country = Array.from(regionInfos.keys()).includes(Config.store.country) ? Config.store.country : 'usa';
+	let country = Config.store.country;
+	if (!regionInfos.has(country)) {
+		country = 'usa';
+	}
+
+	const regionInfo = regionInfos.get(country);
+	if (!regionInfo) {
+		throw new Error(`LogicException could not retrieve region info for ${country}`);
+	}
```

Comment:
I think you can remove this as you already check in lines 11-13?

Reply:
Better approach though. Thanks!

Reply:
> I think you can remove this as you already check in lines 11-13?

I think it needs to be checked anyway or the TypeScript compiler will correctly throw an error that regionInfo may be undefined, e.g. if usa was renamed or removed.

Reply:
Based on that assumption, I suppose you could do the opposite then and do one call.

```typescript
	if (!regionInfo) {
		country = 'usa';
	}
```

Reply:
I may not be following, but the only way this will work without 2 if conditions is if we silence the typescript compiler, e.g.
```ts
function getRegionInfo(): NvidiaRegionInfo {
	let regionInfo = regionInfos.get(Config.store.country);
	if (!regionInfo) {
		regionInfo = regionInfos.get('usa')
	}

	return regionInfo!;
}
```
It has no way to tell that `usa` will always exist in the `regionInfos` map, and it's IMO correct to behave that way.

Reply:
Ah, got it. No problem. I haven't pull your branch, so I guess I haven't seen what the compiler spits out yet. Makes sense to me!

Reply:
Ah. I see now. I come from a different language so the idiosyncrasies are still catching up.

I was thinking usually when you call `T.get()` or us `[]`, if it can't find the object, it'll throw an exception. I see that `Map.get()` in TS will just return an `undefined` object, rather than throw.

My assumption is that we would do something like this:

```
function getRegionInfo(): NvidiaRegionInfo {
	const regionInfo = regionInfos.get(Config.store.country);
	if (!regionInfo) {
		return regionInfos.get('usa')
	}

	return regionInfo;
}
```

But I see the `TypeAssertion` error now.

---

Below isn't a request to change, just some feedback would be nice for a learning experience on my end if you didn't mind!

Would something like this run fine?

```typescript
function getRegionInfo(): NvidiaRegionInfo {
	const regionInfo = regionInfos.get(Config.store.country);
	if (!regionInfo) {
		return <NvidiaRegionInfo>regionInfos.get('usa')
	}

	return regionInfo;
}
```

It compiles, but as long as we know `'usa'` is a provided country, we could. This could still be dangerous as `'usa'` may "suddenly" disappear.

I suppose in that case we could create a wrapper function around `Map.get()` for getting a default `NvidiaRegionInfo` and avoid altogether!

Reply:
That would work, TS allows type coercion so long as the type is legal. Another way to write that would be `regionInfos.get('usa') as NvidiaRegionInfo`, or `regionInfos.get('usa')!` to tell the compiler it's definitely defined. The last one would be most applicable to this example, type coercion is only needed to tell the compiler the result is a one of multiple defined types that are valid.

Reply:
Got it. So best case is to use the implementation you had, but next best is the `regionInfos.get('usa')!`. Makes sense to me. I suppose it lacks a keyword [`unsafe`](https://doc.rust-lang.org/book/ch19-01-unsafe-rust.html) :wink: 

Thanks for the explanation!